applications/ConvectionDiffusionApplication/symbolic_generation/convection_diffusion_explicit/generate_convection_diffusion_explicit_element.py

```diff
@@ -0,0 +1,261 @@
+from KratosMultiphysics import *
+from KratosMultiphysics.sympy_fe_utilities import *
+from sympy import *
+
+def DefineShapeFunctionsMatrix(dim, n_nodes, n_gauss):
+    mat_N = DefineMatrix('mat_N', n_gauss, n_nodes)
+    if dim == 2:
+        if n_gauss == 1:
+            mat_N[0,0] = 1.0 / 3.0
+            mat_N[0,1] = 1.0 / 3.0
+            mat_N[0,2] = 1.0 / 3.0
+        elif n_gauss == 3:
+            mat_N[0,0] = 2.0 / 3.0
+            mat_N[0,1] = 1.0 / 6.0
+            mat_N[0,2] = 1.0 / 6.0
+            mat_N[1,0] = 1.0 / 6.0
+            mat_N[1,1] = 2.0 / 3.0
+            mat_N[1,2] = 1.0 / 6.0
+            mat_N[2,0] = 1.0 / 6.0
+            mat_N[2,1] = 1.0 / 6.0
+            mat_N[2,2] = 2.0 / 3.0
+        else:
+            err_msg = "Invalid quadrature for dimension " + str(dim) + " and number of Gauss points " + str(n_gauss) + "."
+    elif dim == 3:
+        if n_gauss == 1:
+            mat_N[0,0] = 1.0 / 4.0
+            mat_N[0,1] = 1.0 / 4.0
+            mat_N[0,2] = 1.0 / 4.0
+            mat_N[0,3] = 1.0 / 4.0
+        elif n_gauss == 4:
+            mat_N[0,0] = 0.58541020
+            mat_N[0,1] = 0.13819660
+            mat_N[0,2] = 0.13819660
+            mat_N[0,3] = 0.13819660
+            mat_N[1,0] = 0.13819660
+            mat_N[1,1] = 0.58541020
+            mat_N[1,2] = 0.13819660
+            mat_N[1,3] = 0.13819660
+            mat_N[2,0] = 0.13819660
+            mat_N[2,1] = 0.13819660
+            mat_N[2,2] = 0.58541020
+            mat_N[2,3] = 0.13819660
+            mat_N[3,0] = 0.13819660
+            mat_N[3,1] = 0.13819660
+            mat_N[3,2] = 0.13819660
+            mat_N[3,3] = 0.58541020
+        else:
+            err_msg = "Invalid quadrature for dimension " + str(dim) + " and number of Gauss points " + str(n_gauss) + "."
+    else:
+        err_msg = "Invalid dimension " + str(dim) + "."
+        raise Exception(err_msg)
+
+    return mat_N
+
+# Symbolic generation settings
+do_simplifications = False
+dim_to_compute = "Both"             # Spatial dimensions to compute. Options:  "2D","3D","Both"
+ASGS_stabilization = True           # Consider ASGS stabilization terms
+dynamic_subscales = True            # Consider subscale dynamic
+OSS_stabilization = True            # Requires ASGS stabilization to be true
```

Comment:
This could lead to misunderstandings in case a non-experienced user uses the generator. IMO The `ASGS_stabilization` and the `OSS_stabilization` can't be both `True` at the same time.